articles/how-to-use-an-api-in-a-vuejs-project/index.md

```diff
@@ -0,0 +1,345 @@
+In this modern era, the majority of developers are familiar with the language JavaScript and that has led to most web applications having JavaScript as one of the programming languages used in building them. To make the building process easier for developers, frameworks were built with the concept of creating a compiled version of JavaScript that adopts one programming architecture or is dynamic and allows programmers to build their applications on this framework. With this concept in mind came Vue.js.
+
+## What is Vue.js
+
+Vue.js is a progressive framework for JavaScript which is used to build user interfaces. You can read more about it [here](https://vuejs.org/v2/guide/index.html).
+
+When developing some projects using Vue.js, there will be a need to fetch or consume data from an API. This will make the front-end of the application interact with the backend of the application. The fetched data can be consumed on the front-end of the application.
+
+## What is an API?
+
+API stands for Application Programming Interface, which is a set of definitions or protocols that allow software programs to communicate and share data with each other. It is more of a software intermediary.
+
+## Prerequisite
+
+To understand and follow this article, you will need:
+
+- [Node.js](https://nodejs.org/en/download/) installed on your computer.
+
+- Have Vue.js installed on your computer. If you don't already have it installed, click [here](https://vuejs.org/v2/guide/installation.html) to do so.
+
+- Understand key concepts in Vue.js. You can learn more about Vue.js [here](https://vuejs.org/v2/guide/index.html) .
+
+## Outline
+
+1. Using Axios to consume an API
+2. Using Fetch API method
+3. Using APIs on Vuex
+4. Conclusion
+
+There are two methods of consuming/using an API in a Vue.js project. These methods are:
+
+## Using Axios to consume APIs
+
+[Axios](https://www.npmjs.com/package/axios) is a promise-based HTTP client which makes it appropriate for fetching data during server-side rendering. It works on both browser and Node apps.
+
+**Axios Installation**
+
+To use Axios on your project, you need to first install it. This can be done in two ways:
+
+- By using `npm`: a default package manager for the JavaScript runtime environment Node.js.
+
+- By using `yarn`: a package manager that doubles down as project manager. It is compactible with the npm registry and has the same set of features.
+
+With npm:
+
+```bash
+npm i axios
+```
+
+With yarn:
+
+```bash
+yarn add axios
+```
+
+Next, you import axios in your `src/main.js` file
+
+```JavaScript
+    import axios from 'axios';
+    Vue.prototype.$http = axios;
+```
```

Comment:
Before proceeding with this, I think it would be helpful if you instruct the readers to create a vue project using the cli.

Else, you can provide a link to [documentation](https://cli.vuejs.org/guide/creating-a-project.html) that goes over how to create a vue project.